kernel/src/tbfheader.rs

```diff
@@ -55,20 +122,135 @@ crate struct TbfHeaderV2Main {
 ///
 /// There can be multiple (or zero) flash regions defined, so this is its own
 /// struct.
-#[repr(C)]
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, Default)]
 crate struct TbfHeaderV2WriteableFlashRegion {
     writeable_flash_region_offset: u32,
     writeable_flash_region_size: u32,
 }
 
+// Conversion functions from slices to the various TBF fields.
+
+impl core::convert::TryFrom<&[u8]> for TbfHeaderV2Base {
```

Comment:
I don't think I can get a slice with a fixed size type for the conversion. That is, with `TryFrom<&[u8; 16]>`, `let tbf_header_base: TbfHeaderV2Base = header.get(0..16).ok_or(TbfParseError::NotEnoughFlash)?.try_into()?;` gives:

```
error[E0277]: the trait bound `tbfheader::TbfHeaderV2Base: core::convert::From<&[u8]>` is not satisfied
   --> /Users/bradjc/git/tock/kernel/src/tbfheader.rs:417:108
    |
417 |             let tbf_header_base: TbfHeaderV2Base = header.get(0..16).ok_or(TbfParseError::NotEnoughFlash)?.try_into()?;
    |                                                                                                            ^^^^^^^^ the trait `core::convert::From<&[u8]>` is not implemented for `tbfheader::TbfHeaderV2Base`
    |
    = note: required because of the requirements on the impl of `core::convert::Into<tbfheader::TbfHeaderV2Base>` for `&[u8]`
    = note: required because of the requirements on the impl of `core::convert::TryFrom<&[u8]>` for `tbfheader::TbfHeaderV2Base`
    = note: required because of the requirements on the impl of `core::convert::TryInto<tbfheader::TbfHeaderV2Base>` for `&[u8]`
```